src/utils/hocs/protectedRoute.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { ComponentType, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import TObjectLiteral from 'types/TObjectLiteral';
+
+import { selectLogInFailed } from 'store/userProfile/selectors';
+
+import history from 'utils/history';
+
+const ProtectedRoute = (WrappedComponent: ComponentType<TObjectLiteral>) => function Comp(props: TObjectLiteral) {
```

Comment:
🧐
Можно вторую стрелку анонимную передать, на твой вкус
```suggestion
const ProtectedRoute = (WrappedComponent: ComponentType<TObjectLiteral>) => (props: TObjectLiteral) => {
```

Reply:
тогда пишет такую ошибку: `React Hook "useSelector" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function.`